contract_manager/src/chains.ts

```diff
@@ -858,3 +860,84 @@ export class TonChain extends Chain {
     return Number(balance) / 10 ** 9;
   }
 }
+
+export class NearChain extends Chain {
+  static type = "NearChain";
+  networkId: string;
+
+  constructor(
+    id: string,
+    mainnet: boolean,
+    wormholeChainName: string,
+    nativeToken: TokenId | undefined
+  ) {
+    super(id, mainnet, wormholeChainName, nativeToken);
+    this.networkId = this.mainnet ? "mainnet" : "testnet";
+  }
+
+  static fromJson(parsed: ChainConfig): NearChain {
+    if (parsed.type !== NearChain.type) throw new Error("Invalid type");
+    return new NearChain(
+      parsed.id,
+      parsed.mainnet,
+      parsed.wormholeChainName,
+      parsed.nativeToken
+    );
+  }
+
+  getType(): string {
+    return TonChain.type;
+  }
+
+  toJson(): KeyValueConfig {
+    return {
+      id: this.id,
+      wormholeChainName: this.wormholeChainName,
+      mainnet: this.mainnet,
+      type: TonChain.type,
+    };
+  }
+
+  /**
+   * Returns the payload for a governance contract upgrade instruction for contracts deployed on this chain
+   * @param codeHash hex string of the 32 byte code hash for the new contract without the 0x prefix
+   */
+  generateGovernanceUpgradePayload(codeHash: string): Buffer {
+    return new UpgradeContract256Bit(this.wormholeChainName, codeHash).encode();
+  }
+
+  async getAccountAddress(privateKey: PrivateKey): Promise<string> {
+    return Buffer.from(
+      Ed25519Keypair.fromSecretKey(Buffer.from(privateKey, "hex"))
+        .getPublicKey()
+        .toRawBytes()
+    ).toString("hex");
+  }
+
+  async getAccountBalance(privateKey: PrivateKey): Promise<number> {
+    const accountId = await this.getAccountAddress(privateKey);
+    const account = await this.getNearAccount(accountId);
+    const balance = await account.getAccountBalance();
+    return Number(balance.available) / 1e24;
+  }
+
+  async getNearAccount(
+    accountId: string,
+    senderPrivateKey?: PrivateKey
+  ): Promise<nearAPI.Account> {
+    const keyStore = new nearAPI.keyStores.InMemoryKeyStore();
+    if (typeof senderPrivateKey !== "undefined") {
+      const key = bs58.encode(Buffer.from(senderPrivateKey, "hex"));
+      const keyPair = nearAPI.KeyPair.fromString(key);
+      const address = await this.getAccountAddress(senderPrivateKey);
+      await keyStore.setKey(this.networkId, address, keyPair);
+    }
+    const connectionConfig = {
+      networkId: this.networkId,
+      keyStore,
+      nodeUrl: `https://rpc.${this.networkId}.near.org`,
```

Comment:
couldn't we add this to the chain configs? (just to make changes to it easier)

Reply:
Moved to yaml configs.